blocks/bricks/recurrent/base.py

```diff
@@ -46,6 +46,9 @@ def initial_states(self, batch_size, *args, **kwargs):
             The keyword arguments of the application call.
 
         """
+        if not hasattr(self, 'apply') or not self.apply.states:
+            return
+
```

Comment:
@Beronx86 , checking `apply.states` in `BaseRecurrent.initial_states` is not a solution. There are quite a few places in Blocks-dependent code where `initial_states` method is overloaded. Instead, like in your previous solution, `initial_states` should not be called if `application` does not have `states`. Can you please revert back to the previous version of your fix?